### DISCUSSION TITLE
Use of media devices and desktop sharing.

### DISCUSSION THREAD

**Discussion by rdwnsjjd** [tauri-apps/tauri#5572] [Q&A] [answered]
It is possible to use  `navigator.mediaDevices.getUserMedia` as we use it in browser to get camera-microphone or desktop sharing in tauri?
I printed my navigator on console and it doesn't have any `mediaDevice` method or value. how can I do this in tauri?

**Comment by armand-colin** [comment, marked as answer]
Yes it is possible. Are you using macOS? If so, you will need an `Info.plist` file to give os accesses to your program.
Check out this issue https://github.com/tauri-apps/tauri/issues/2600. The guy has written down his plist file and it seems correct.
Concerning screen sharing, it works on windows and it just became available with the latest version of macOS (Ventura 13).

**Reply by rdwnsjjd** [reply]
no I use Linux. it is possible on Linux?

**Reply by armand-colin** [reply]
If I remember well, the webview system used by tauri on Linux does not support WebRTC atm, and so no it does not work

**Reply by rdwnsjjd** [reply]
so it might be not possible with others like electron right?

**Reply by armand-colin** [reply]
Electron packages a Chromium no matter what os you use I think, so it may be possible on electron

**Reply by rdwnsjjd** [reply]
tnx🌱

**Reply by thenbe** [reply]
@rdwnsjjd see here for preliminary linux support: https://github.com/tauri-apps/tauri/discussions/8426#discussioncomment-8268622

**Comment by jclaessens97** [comment]
@armand-colin I got `userMedia` device working with the `Info.plist`, but only for the built-in webcam.
Whenever I want to select another capture device (e.g. OBS Virtual Cam or GoPro Cam) I get following error.

![image](https://user-images.githubusercontent.com/10177148/221405113-8324314e-61e1-449f-836c-b5029a0c2ec0.png)

I assume that in the background this relates to another permission issue, but I'm not entirely sure about it. In browser it works just fine.

**Reply by jclaessens97** [reply]
Correction, it only works in chrome, on Safari I can only get access to the built-in webcam.

**Reply by Manubi** [reply]
@jclaessens97 where you able to connect to other devices on macos? The problem is that safari doesn't support several webapis. Did you also try to connect via bluetooth?

**Comment by matthiasg** [comment]
I got this working with Jitsi, but there is no prompt for screen sharing yet ?

**Comment by bhagwanZaki** [comment]
I am unable to start the microphone in windows, when i enumerate the devices I can see the microphone list but when i try get media it is throwing me "NotReadableError: Could not start audio source".
